features/preferences/impl/src/main/kotlin/io/element/android/features/preferences/impl/notifications/edit/EditDefaultNotificationSettingView.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.features.preferences.impl.notifications.edit
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.selection.selectableGroup
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import io.element.android.libraries.designsystem.components.preferences.PreferenceCategory
+import io.element.android.libraries.designsystem.components.preferences.PreferenceView
+import io.element.android.libraries.matrix.api.room.RoomNotificationMode
+import io.element.android.libraries.ui.strings.CommonStrings
+
+@Composable
+fun EditDefaultNotificationSettingView(
+    state: EditDefaultNotificationSettingState,
+    onBackPressed: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+
+    val title = if(state.isOneToOne) {
+        CommonStrings.screen_notification_settings_direct_chats
+    } else {
+        CommonStrings.screen_notification_settings_group_chats
+    }
+    PreferenceView(
+        modifier = modifier,
+        onBackPressed = onBackPressed,
+        title = stringResource(id = title)
+    ) {
+
+        val validModes = listOf(RoomNotificationMode.ALL_MESSAGES, RoomNotificationMode.MENTIONS_AND_KEYWORDS_ONLY)
```

Comment:
No silent mode here?

Reply:
No, `ALL_MESSAGES` and `MENTIONS_AND_KEYWORDS_ONLY ` are the only valid values for global defaults on all the clients. Even on wed which has "Off" "On" and "Nosiey" in its legacy settings, "Off" maps to `MENTIONS_AND_KEYWORDS_ONLY` and "On" and "Nosiey" map to `ALL_MESSAGES`.

Reply:
Will add a code comment to that effect.